### PR TITLE
feat(HACBS-1829): add ability to push sha or timestamp tags

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

File: catalog/task/push-snapshot/0.5/push-snapshot.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          # note: Inspection might fail on empty repos, hence `|| true`
+          destination_digest=$(
+            skopeo inspect \
+            --format '{{.Digest}}' \
+            "docker://${data[repository]}:${data[tag]}" 2>/dev/null || true)
+          if [[ "$destination_digest" != "$source_digest" || -z "$destination_digest" ]]
+          then
+            push_image "${data[name]}" "${data[containerImage]}" "${data[repository]}" "${data[tag]}"
+            if [ $(params.addTimestampTag) = true ] ; then
+              timestamp="TODO"
```
Comment (johnbieren) [inline catalog/task/push-snapshot/0.5/push-snapshot.yaml:78]: This should be modified once the timestamp format is decided on. The PR will remain a draft until then

File: catalog/task/push-snapshot/0.5/push-snapshot.yaml [hunk 2]
```diff
@@ -53,14 +73,21 @@ spec:
...
+              push_image "${data[name]}" "${data[containerImage]}" "${data[repository]}" "$timestamp"
+            fi
+            if [ $(params.addShaTag) = true ] ; then
+              if [[ "${data[containerImage]}" == *"@sha256"* && \
+                $(echo "${data[containerImage]}" | tr -cd ':' | wc -c) -eq 1 ]]
+              then
+                sha=$(echo "${data[containerImage]}" | cut -d ':' -f 2)
+                push_image "${data[name]}" "${data[containerImage]}" "${data[repository]}" "$sha"
+              else
+                printf 'Asked to create sha based tag, but no sha found in %s\n' "${data[containerImage]}"
```
Comment (scoheb) [inline catalog/task/push-snapshot/0.5/push-snapshot.yaml:88]: should this surface as an error?

File: catalog/task/push-snapshot/0.5/push-snapshot.yaml (hunk 1 above)
Comment (scoheb) [reply catalog/task/push-snapshot/0.5/push-snapshot.yaml:78]: Being discussed in https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1677688042765809

File: catalog/task/push-snapshot/0.5/push-snapshot.yaml (hunk 2 above)
Comment (johnbieren) [reply catalog/task/push-snapshot/0.5/push-snapshot.yaml:88]: I can error out in this case if you think that makes the most sense. Not sure if we have a snapshot of X images and 1 is bad, we still want to push the X-1 or not

Comment (scoheb) [reply catalog/task/push-snapshot/0.5/push-snapshot.yaml:88]: I think for the time being, it is better to error out. @davidmogar ?

Comment (johnbieren) [reply catalog/task/push-snapshot/0.5/push-snapshot.yaml:88]: @davidmogar thoughts? One thing is the images prior to the one with the potential issue (if any) would already have been pushed, `exit 1` or not

Comment (davidmogar) [reply catalog/task/push-snapshot/0.5/push-snapshot.yaml:88]: We've been talking about making our pipelines more robust and part of it was to recover from an error on the next run. If this is an error, it should exit with 1 and be recovered in the next run.

Comment (johnbieren) [reply catalog/task/push-snapshot/0.5/push-snapshot.yaml:88]: Added `exit 1`

**Review by davidmogar — APPROVED** [review]
/lgtm

**Pull request merged**